### DISCUSSION TITLE
[Tracing, Metrics] Perfetto Integration / Export

### DISCUSSION THREAD

**Discussion by bsarden** [open-telemetry/opentelemetry-specification#2657] [General]
I'm evaluating the differences between perfetto and the open-telemetry ecosystem, and it seems like there could be a lot of value in reusing the perfetto ui for viewing otel metrics+span sdk's. Specifically, being able to reuse the [trace visualization](https://perfetto.dev/docs/#trace-visualization) and the [trace processor](https://perfetto.dev/docs/analysis/trace-processor) to have a cohesive story for running SQL expressions on traces.

Besides otel lacking the profiling support, being able to reuse the span / metrics sdk as a drop-in replacement for perfetto's TRACE_EVENT and TRACE_COUNTER macros seems like it may be worthwhile to others as well. Is this something other members in the community are interested in as well?

Naively, it seems like perfetto was designed for tracing on-device behavior while otel focuses more on distributed tracing currently. Maybe there is no overlap here, but figured it would be worth asking about :).

For tracking, I originally brought this up in the `opentelemetry-cpp` repository [here](https://github.com/open-telemetry/opentelemetry-cpp/discussions/1483) (since perfetto is largely a C/C++ tool), but moving the conversation here.

**Comment by JoeLoser** [comment]
@bsarden I'd be interested in something like this for combining OpenTelemetry tracing to Perfetto.  Do you have anything to share or review?  I'd be happy to be a reviewer.

**Comment by anoopknayak** [comment]
I just wanted to bump on this conversation since it hasn't been updated in a while. Has this been already solved/discussed? If not, I'd like to contribute as well.
